tests/test_pipx_metadata_file.py

```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+import pytest  # type: ignore
+
+from helpers import assert_not_in_virtualenv, run_pipx_cli
+import pipx.constants
+from pipx.pipx_metadata_file import PipxMetadata, PackageInfo
+from pipx.util import PipxError
+
+assert_not_in_virtualenv()
```

Comment:
This shouldn't be needed here. It should only be run once per pytest session which is a TODO. Right now > 1 files run this function because I copy+pasted by accident, but just one file running it is sufficient.

Reply:
Is every file run during any pytest execution? i.e. does pytest do its full test discovery routine even if you only give it an explicit test to run? I suppose it must as it always picks up stuff in `conftest`. I feel unsure about asserts happening at the module level, though.

This seems to be the pytest-y way to do it: https://stackoverflow.com/a/17844938/2700168

Reply:
Ah the `@pytest.fixture(scope="session", autouse=True)` looks good.  

I can include it in this PR, or should I save that for later, and here just remove the extra `assert_not_in_virtualenv()`?

Reply:
>  I suppose it must as it always picks up stuff in conftest. 

Agreed, that is the assumption here.

> I feel unsure about asserts happening at the module level, though.

Why? The module level code must be executed to run the test.

Reply:
> Ah the `@pytest.fixture(scope="session", autouse=True)` looks good.
> 
> I can include it in this PR, or should I save that for later, and here just remove the extra `assert_not_in_virtualenv()`?

It can be added in a different PR.